exercises/simple-linked-list/linked_list_test.go

```diff
@@ -0,0 +1,191 @@
+// API:
+//
+// type Element struct {
+//  data int
+//  next *Element
+// }
+//
+// type List struct {
+//  head *Element
+//  size int
+// }
+//
+// func New([]int) *List
+// func (*List) Size() int
+// func (*List) Push(int)
+// func (*List) Pop() error
+// func (*List) Array() []int
+// func (*List) Reverse() *List
+
+package linkedlist
+
+import (
+	"reflect"
+	"testing"
+)
+
+var array1To10 = []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+
+func TestEmptyListHasSizeZero(t *testing.T) {
+	list := New([]int{})
```

Comment:
how about `New(nil)` too ?

Reply:
should we expect the solution to handle a nil input? In that case, should I add tests for all empty-list tests but with nil input?

Reply:
I would expect it to behave similar since we didn't supply any element from the point of linked list.